pennylane_lightning/lightning_qubit.py

```diff
@@ -462,36 +454,36 @@ def batch_vjp(
             use_device_state (bool): use current device state to initialize. A forward pass of the same
                 circuit should be the last thing the device has executed. If a ``starting_state`` is
                 provided, that takes precedence.
-
         Returns:
-            tuple[List[array or None], List[tensor_like or None]]: A tuple containing a list
-            of adjoint-jacobians and a list of vector-Jacobian products. ``None`` elements corresponds
-            to tapes with no trainable parameters.
+            The processing function required to compute the vector-Jacobian products of a batch of tapes.
         """
-        vjps = []
-        jacs = []
+        fns = []
 
         # Loop through the tapes and dys vector
         for tape, dy in zip(tapes, dys):
-            jac, vjp = self.vector_jacobian_product(
-                tape,
-                dy,
-                starting_state=starting_state,
-                use_device_state=use_device_state,
+            fn = self.vjp(
+                tape, dy, starting_state=starting_state, use_device_state=use_device_state
             )
-            if vjp is None:
-                if reduction == "append":
-                    vjps.append(None)
-                    jacs.append(jac)
-                continue
-            if isinstance(reduction, str):
-                getattr(vjps, reduction)(vjp)
-                getattr(jacs, reduction)(jac)
-            elif callable(reduction):
-                reduction(vjps, vjp)
-                reduction(jacs, jac)
-
-        return jacs, vjps
+            fns.append(fn)
+
+        def processing_fns(tapes):
+            vjps = []
+            for t, f in zip(tapes, fns):
+                vjp = f(t)
+
+                if vjp is None:
+                    if reduction == "append":
+                        vjps.append(None)
+                    continue
+
+                if isinstance(reduction, str):
+                    getattr(vjps, reduction)(vjp)
+                elif callable(reduction):
+                    reduction(vjps, vjp)
```

Comment:
I'm a little unsure about the structure here. Can you elaborate a little on the intention of what is to happen?

Reply:
I didn't get it when I saw this at first [here](https://github.com/PennyLaneAI/pennylane/blob/5a28983fc7bd950cde8a4014e54261fef4b54293/pennylane/gradients/vjp.py#L341). However, it seems the end user may pass the argument as `list.extend`, `"extend"` or `"append"` (there are tests covering this part [here](https://github.com/PennyLaneAI/pennylane-lightning/blob/e4a21c51379cbe6350042a6fb94c5d8e615f5f31/tests/test_vjp.py#L576))

Reply:
This particular detail in the PennyLane VJP implementation is a bit ugly, and exists only for one reason:

- Autograd and JAX like their VJPs appended to a list, resulting in a nested list
- TensorFlow and Torch like their VJPs extended into a flat list

Being able to specify how the reduction takes place just avoided us duplicating the function.

Reply:
But if you don't need this here, I think you can avoid this

Reply:
Fair enough! Thanks for the clarification. As it is not harmful, I think we can keep it to be consistent with the PL VJP implementation.